src/exec.cpp

```diff
@@ -979,7 +982,23 @@ void Executor::raise(const std::string &exception, const std::string &info)
         ip = callstack.top();
         callstack.pop();
     }
-    fprintf(stderr, "unhandled exception %s (%s)\n", exception.c_str(), info.c_str());
+    fprintf(stderr, "unhandled exception %s (%s) at address %lu\n", exception.c_str(), info.c_str(), eip);
+    
+    if (debug != nullptr) {
+        auto line = debug->line_numbers.find(eip);
+        if (line == debug->line_numbers.end()) {
```

Comment:
The nesting here seems too deep. You should be able to do this with just a single `while` statement, and not an `if/for` nested block.